lib/core/src/restore.rs

```diff
@@ -0,0 +1,1164 @@
+//! This module provides functionality for restoring the swap tx IDs from onchain data
+
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Result};
+use log::{error, info};
+use lwk_wollet::elements::Txid;
+use lwk_wollet::WalletTx;
+use sdk_common::bitcoin::hashes::hex::ToHex;
+
+use crate::prelude::*;
+use crate::restore::immutable::*;
+
+/// A map of all our known LWK onchain txs, indexed by tx ID. Essentially our own cache of the LWK txs.
+pub(crate) struct TxMap {
+    outgoing_tx_map: HashMap<Txid, WalletTx>,
+    incoming_tx_map: HashMap<Txid, WalletTx>,
+}
+impl TxMap {
+    pub(crate) fn from_raw_tx_map(raw_tx_map: HashMap<Txid, WalletTx>) -> Self {
+        let (outgoing_tx_map, incoming_tx_map): (HashMap<Txid, WalletTx>, HashMap<Txid, WalletTx>) =
+            raw_tx_map
+                .into_iter()
+                .partition(|(_, tx)| tx.balance.values().sum::<i64>() < 0);
+
+        Self {
+            outgoing_tx_map,
+            incoming_tx_map,
+        }
+    }
+}
+
+trait PartialSwapState {
```

Comment:
> Why do we need a trait here compare to only a function for each swap?

I did it that way so the function `get_partial_state` can have the docs in one place, applicable to all instances. IMO it's important to point out why its a partial state and what it means.

> Also IMO it would be more clear to call the function `restore_state()`

Restore implies it's storing it somewhere, which it doesn't. I renamed it to `derive_partial_state`:  9566f7b87390b824b1e2015221b22156eb598054